test/test_scandir.py

```diff
@@ -250,6 +250,23 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_os_walk_unicode_handling(self):
+        if sys.platform == 'win32':
+            return self.skipTest("Not windows")
+        subdir = os.path.join(TEST_PATH, 'walk').encode('utf-8')
```

Comment:
It doesn't seem ok to encode TEST_PATH on Python 2 since it's already a byte string. You should use os.fsencode(), not hardcode UTF-8. Encoding TEST_PATH can fail on Python 3. Test with an ASCII locale encoding (ex: "LC_ALL=C") and a non-ASCII directory.